cpg_infra/billing_aggregator/aggregate/utils.py

```diff
@@ -464,6 +469,7 @@ async def get_finished_batches_for_date(
 async def get_jobs_for_batch(
     batch_id: int,
     token: str,
+    limit: int | None = None,
```

Comment:
These two limits are for different things, this limit is the _number of jobs to get per request_, not the total number of jobs before we back out. I think you're better to do the query_on_batch check before it gets to here, even write a specific function that takes in a batch_record, and returns True if we guess it to be a hail query on batch, batch (name is blank, and n_jobs > 10k ish).

Reply:
Removed that, taking care of batch queries prior, checking name attribute.